From 1badb810c31321bbff53486d2d0cf1bae45e9c0c Mon Sep 17 00:00:00 2001
From: Mike Looijmans <mike.looijmans@topic.nl>
Date: Wed, 3 Jun 2020 10:35:03 +0200
Subject: [PATCH] board: topic-miamiplusmp: Set DDR to 1600MT/s

Clock the DDR at 800 MHz and set the timings accordingly. This reduces the
power consumption of the DDR.
---
 .../zynqmp-topic-miamiplusmp/psu_init_gpl.c   | 100 +++++++++---------
 1 file changed, 50 insertions(+), 50 deletions(-)

diff --git a/board/topic/zynqmp/zynqmp-topic-miamiplusmp/psu_init_gpl.c b/board/topic/zynqmp/zynqmp-topic-miamiplusmp/psu_init_gpl.c
index 35dc6d8489..fd5edafa67 100644
--- a/board/topic/zynqmp/zynqmp-topic-miamiplusmp/psu_init_gpl.c
+++ b/board/topic/zynqmp/zynqmp-topic-miamiplusmp/psu_init_gpl.c
@@ -34,13 +34,13 @@ static unsigned long psu_pll_init_data(void)
 	psu_mask_write(0xFD1A0020, 0x00000008U, 0x00000000U);
 	psu_mask_write(0xFD1A0048, 0x00003F00U, 0x00000300U);
 	psu_mask_write(0xFD1A0030, 0xFE7FEDEFU, 0x7E4B0C62U);
-	psu_mask_write(0xFD1A002C, 0x00717F00U, 0x00014000U);
+	psu_mask_write(0xFD1A002C, 0x00717F00U, 0x00014800U);
 	psu_mask_write(0xFD1A002C, 0x00000008U, 0x00000008U);
 	psu_mask_write(0xFD1A002C, 0x00000001U, 0x00000001U);
 	psu_mask_write(0xFD1A002C, 0x00000001U, 0x00000000U);
 	mask_poll(0xFD1A0044, 0x00000002U);
 	psu_mask_write(0xFD1A002C, 0x00000008U, 0x00000000U);
-	psu_mask_write(0xFD1A004C, 0x00003F00U, 0x00000200U);
+	psu_mask_write(0xFD1A004C, 0x00003F00U, 0x00000300U);
 	psu_mask_write(0xFD1A003C, 0xFE7FEDEFU, 0x7E4B0C62U);
 	psu_mask_write(0xFD1A0038, 0x00717F00U, 0x00014800U);
 	psu_mask_write(0xFD1A0038, 0x00000008U, 0x00000008U);
@@ -88,12 +88,12 @@ static unsigned long psu_clock_init_data(void)
 	psu_mask_write(0xFD1A00B4, 0x01003F07U, 0x01000200U);
 	psu_mask_write(0xFD1A0060, 0x03003F07U, 0x03000100U);
 	psu_mask_write(0xFD1A0068, 0x01003F07U, 0x01000200U);
-	psu_mask_write(0xFD1A0080, 0x00003F07U, 0x00000200U);
+	psu_mask_write(0xFD1A0080, 0x00003F07U, 0x00000300U);
 	psu_mask_write(0xFD1A0084, 0x07003F07U, 0x07000202U);
 	psu_mask_write(0xFD1A00B8, 0x01003F07U, 0x01000200U);
 	psu_mask_write(0xFD1A00BC, 0x01003F07U, 0x01000200U);
-	psu_mask_write(0xFD1A00C0, 0x01003F07U, 0x01000203U);
-	psu_mask_write(0xFD1A00C4, 0x01003F07U, 0x01000502U);
+	psu_mask_write(0xFD1A00C0, 0x01003F07U, 0x01000300U);
+	psu_mask_write(0xFD1A00C4, 0x01003F07U, 0x01000C00U);
 	psu_mask_write(0xFD1A00F8, 0x00003F07U, 0x00000200U);
 	psu_mask_write(0xFF180380, 0x000000FFU, 0x00000000U);
 	psu_mask_write(0xFD610100, 0x00000001U, 0x00000000U);
@@ -111,47 +111,47 @@ static unsigned long psu_ddr_init_data(void)
 	psu_mask_write(0xFD070020, 0x000003F3U, 0x00000200U);
 	psu_mask_write(0xFD070024, 0xFFFFFFFFU, 0x00800000U);
 	psu_mask_write(0xFD070030, 0x0000007FU, 0x00000000U);
-	psu_mask_write(0xFD070034, 0x00FFFF1FU, 0x00408410U);
+	psu_mask_write(0xFD070034, 0x00FFFF1FU, 0x00406310U);
 	psu_mask_write(0xFD070050, 0x00F1F1F4U, 0x00210000U);
 	psu_mask_write(0xFD070054, 0x0FFF0FFFU, 0x00000000U);
 	psu_mask_write(0xFD070060, 0x00000073U, 0x00000001U);
-	psu_mask_write(0xFD070064, 0x0FFF83FFU, 0x008180BBU);
+	psu_mask_write(0xFD070064, 0x0FFF83FFU, 0x0061808CU);
 	psu_mask_write(0xFD070070, 0x00000017U, 0x00000010U);
 	psu_mask_write(0xFD070074, 0x00000003U, 0x00000000U);
 	psu_mask_write(0xFD0700C4, 0x3F000391U, 0x10000200U);
-	psu_mask_write(0xFD0700C8, 0x01FF1F3FU, 0x0040051FU);
-	psu_mask_write(0xFD0700D0, 0xC3FF0FFFU, 0x00020106U);
+	psu_mask_write(0xFD0700C8, 0x01FF1F3FU, 0x0030051FU);
+	psu_mask_write(0xFD0700D0, 0xC3FF0FFFU, 0x000200C5U);
 	psu_mask_write(0xFD0700D4, 0x01FF7F0FU, 0x00020000U);
-	psu_mask_write(0xFD0700D8, 0x0000FF0FU, 0x00002305U);
-	psu_mask_write(0xFD0700DC, 0xFFFFFFFFU, 0x07340301U);
-	psu_mask_write(0xFD0700E0, 0xFFFFFFFFU, 0x00200200U);
+	psu_mask_write(0xFD0700D8, 0x0000FF0FU, 0x00001A05U);
+	psu_mask_write(0xFD0700DC, 0xFFFFFFFFU, 0x03140301U);
+	psu_mask_write(0xFD0700E0, 0xFFFFFFFFU, 0x00100000U);
 	psu_mask_write(0xFD0700E4, 0x00FF03FFU, 0x00210004U);
 	psu_mask_write(0xFD0700E8, 0xFFFFFFFFU, 0x000006C0U);
-	psu_mask_write(0xFD0700EC, 0xFFFF0000U, 0x08190000U);
+	psu_mask_write(0xFD0700EC, 0xFFFF0000U, 0x04190000U);
 	psu_mask_write(0xFD0700F0, 0x0000003FU, 0x00000010U);
 	psu_mask_write(0xFD0700F4, 0x00000FFFU, 0x0000066FU);
-	psu_mask_write(0xFD070100, 0x7F3F7F3FU, 0x11102412U);
-	psu_mask_write(0xFD070104, 0x001F1F7FU, 0x0004041AU);
-	psu_mask_write(0xFD070108, 0x3F3F3F3FU, 0x0708060DU);
+	psu_mask_write(0xFD070100, 0x7F3F7F3FU, 0x0D0E1A0EU);
+	psu_mask_write(0xFD070104, 0x001F1F7FU, 0x00030314U);
+	psu_mask_write(0xFD070108, 0x3F3F3F3FU, 0x0606060BU);
 	psu_mask_write(0xFD07010C, 0x3FF3F3FFU, 0x0050400CU);
-	psu_mask_write(0xFD070110, 0x1F0F0F1FU, 0x08030409U);
-	psu_mask_write(0xFD070114, 0x0F0F3F1FU, 0x06060403U);
-	psu_mask_write(0xFD070118, 0x0F0F000FU, 0x01010004U);
-	psu_mask_write(0xFD07011C, 0x00000F0FU, 0x00000606U);
-	psu_mask_write(0xFD070120, 0x7F7F7F7FU, 0x04040D07U);
-	psu_mask_write(0xFD070124, 0x40070F3FU, 0x0002030BU);
-	psu_mask_write(0xFD07012C, 0x7F1F031FU, 0x1207010EU);
+	psu_mask_write(0xFD070110, 0x1F0F0F1FU, 0x06030307U);
+	psu_mask_write(0xFD070114, 0x0F0F3F1FU, 0x04040302U);
+	psu_mask_write(0xFD070118, 0x0F0F000FU, 0x01010003U);
+	psu_mask_write(0xFD07011C, 0x00000F0FU, 0x00000404U);
+	psu_mask_write(0xFD070120, 0x7F7F7F7FU, 0x04040B06U);
+	psu_mask_write(0xFD070124, 0x40070F3FU, 0x00020309U);
+	psu_mask_write(0xFD07012C, 0x7F1F031FU, 0x0E05010EU);
 	psu_mask_write(0xFD070130, 0x00030F1FU, 0x00020608U);
 	psu_mask_write(0xFD070180, 0xF7FF03FFU, 0x81000040U);
-	psu_mask_write(0xFD070184, 0x3FFFFFFFU, 0x020196E5U);
-	psu_mask_write(0xFD070190, 0x1FBFBF3FU, 0x048C820BU);
+	psu_mask_write(0xFD070184, 0x3FFFFFFFU, 0x0201312CU);
+	psu_mask_write(0xFD070190, 0x1FBFBF3FU, 0x04888208U);
 	psu_mask_write(0xFD070194, 0xF31F0F0FU, 0x00030304U);
 	psu_mask_write(0xFD070198, 0x0FF1F1F1U, 0x07000101U);
 	psu_mask_write(0xFD07019C, 0x000000F1U, 0x00000021U);
 	psu_mask_write(0xFD0701A0, 0xC3FF03FFU, 0x00400003U);
 	psu_mask_write(0xFD0701A4, 0x00FF00FFU, 0x00C800FFU);
 	psu_mask_write(0xFD0701B0, 0x00000007U, 0x00000000U);
-	psu_mask_write(0xFD0701B4, 0x00003F3FU, 0x00000A09U);
+	psu_mask_write(0xFD0701B4, 0x00003F3FU, 0x00000606U);
 	psu_mask_write(0xFD0701C0, 0x00000007U, 0x00000001U);
 	psu_mask_write(0xFD070200, 0x0000001FU, 0x0000001FU);
 	psu_mask_write(0xFD070204, 0x001F1F1FU, 0x001F0909U);
@@ -165,7 +165,7 @@ static unsigned long psu_ddr_init_data(void)
 	psu_mask_write(0xFD070224, 0x0F0F0F0FU, 0x07070707U);
 	psu_mask_write(0xFD070228, 0x0F0F0F0FU, 0x07070707U);
 	psu_mask_write(0xFD07022C, 0x0000000FU, 0x00000007U);
-	psu_mask_write(0xFD070240, 0x0F1F0F7CU, 0x06000604U);
+	psu_mask_write(0xFD070240, 0x0F1F0F7CU, 0x06000600U);
 	psu_mask_write(0xFD070244, 0x00003333U, 0x00000001U);
 	psu_mask_write(0xFD070250, 0x7FFF3F07U, 0x01002001U);
 	psu_mask_write(0xFD070264, 0xFF00FFFFU, 0x08000040U);
@@ -223,34 +223,34 @@ static unsigned long psu_ddr_init_data(void)
 	psu_mask_write(0xFD072190, 0x1FBFBF3FU, 0x07828002U);
 	psu_mask_write(0xFD1A0108, 0x0000000CU, 0x00000000U);
 	psu_mask_write(0xFD080010, 0xFFFFFFFFU, 0x07001E00U);
-	psu_mask_write(0xFD080018, 0xFFFFFFFFU, 0x00F10010U);
+	psu_mask_write(0xFD080018, 0xFFFFFFFFU, 0x00F0BF10U);
 	psu_mask_write(0xFD08001C, 0xFFFFFFFFU, 0x55AA5480U);
 	psu_mask_write(0xFD080024, 0xFFFFFFFFU, 0x010100F4U);
-	psu_mask_write(0xFD080040, 0xFFFFFFFFU, 0x42C21590U);
-	psu_mask_write(0xFD080044, 0xFFFFFFFFU, 0xD05512C0U);
-	psu_mask_write(0xFD080068, 0xFFFFFFFFU, 0x01100000U);
-	psu_mask_write(0xFD080090, 0xFFFFFFFFU, 0x02A04161U);
+	psu_mask_write(0xFD080040, 0xFFFFFFFFU, 0x32019010U);
+	psu_mask_write(0xFD080044, 0xFFFFFFFFU, 0x9C400E10U);
+	psu_mask_write(0xFD080068, 0xFFFFFFFFU, 0x02120000U);
+	psu_mask_write(0xFD080090, 0xFFFFFFFFU, 0x02A040E1U);
 	psu_mask_write(0xFD0800C0, 0xFFFFFFFFU, 0x00000000U);
 	psu_mask_write(0xFD0800C4, 0xFFFFFFFFU, 0x000000E0U);
 	psu_mask_write(0xFD080100, 0xFFFFFFFFU, 0x0800040CU);
-	psu_mask_write(0xFD080110, 0xFFFFFFFFU, 0x07241008U);
-	psu_mask_write(0xFD080114, 0xFFFFFFFFU, 0x28200008U);
-	psu_mask_write(0xFD080118, 0xFFFFFFFFU, 0x000F0300U);
-	psu_mask_write(0xFD08011C, 0xFFFFFFFFU, 0x83000800U);
-	psu_mask_write(0xFD080120, 0xFFFFFFFFU, 0x01762B07U);
-	psu_mask_write(0xFD080124, 0xFFFFFFFFU, 0x00331008U);
-	psu_mask_write(0xFD080128, 0xFFFFFFFFU, 0x00000E10U);
+	psu_mask_write(0xFD080110, 0xFFFFFFFFU, 0x061C0C06U);
+	psu_mask_write(0xFD080114, 0xFFFFFFFFU, 0x281C0008U);
+	psu_mask_write(0xFD080118, 0xFFFFFFFFU, 0x000F0255U);
+	psu_mask_write(0xFD08011C, 0xFFFFFFFFU, 0x82550800U);
+	psu_mask_write(0xFD080120, 0xFFFFFFFFU, 0x01182B05U);
+	psu_mask_write(0xFD080124, 0xFFFFFFFFU, 0x00280C06U);
+	psu_mask_write(0xFD080128, 0xFFFFFFFFU, 0x00000B0CU);
 	psu_mask_write(0xFD080140, 0xFFFFFFFFU, 0x08400020U);
 	psu_mask_write(0xFD080144, 0xFFFFFFFFU, 0x00000C80U);
 	psu_mask_write(0xFD080150, 0xFFFFFFFFU, 0x00000000U);
-	psu_mask_write(0xFD080154, 0xFFFFFFFFU, 0x00000200U);
-	psu_mask_write(0xFD080180, 0xFFFFFFFFU, 0x00000634U);
+	psu_mask_write(0xFD080154, 0xFFFFFFFFU, 0x00000000U);
+	psu_mask_write(0xFD080180, 0xFFFFFFFFU, 0x00000214U);
 	psu_mask_write(0xFD080184, 0xFFFFFFFFU, 0x00000301U);
-	psu_mask_write(0xFD080188, 0xFFFFFFFFU, 0x00000020U);
-	psu_mask_write(0xFD08018C, 0xFFFFFFFFU, 0x00000200U);
+	psu_mask_write(0xFD080188, 0xFFFFFFFFU, 0x00000010U);
+	psu_mask_write(0xFD08018C, 0xFFFFFFFFU, 0x00000000U);
 	psu_mask_write(0xFD080190, 0xFFFFFFFFU, 0x00000000U);
 	psu_mask_write(0xFD080194, 0xFFFFFFFFU, 0x000006C0U);
-	psu_mask_write(0xFD080198, 0xFFFFFFFFU, 0x00000819U);
+	psu_mask_write(0xFD080198, 0xFFFFFFFFU, 0x00000419U);
 	psu_mask_write(0xFD0801AC, 0xFFFFFFFFU, 0x00000000U);
 	psu_mask_write(0xFD0801B0, 0xFFFFFFFFU, 0x0000004DU);
 	psu_mask_write(0xFD0801B4, 0xFFFFFFFFU, 0x00000008U);
@@ -275,7 +275,7 @@ static unsigned long psu_ddr_init_data(void)
 	psu_mask_write(0xFD08055C, 0xFFFFFFFFU, 0x00000000U);
 	psu_mask_write(0xFD080560, 0xFFFFFFFFU, 0x00000000U);
 	psu_mask_write(0xFD080564, 0xFFFFFFFFU, 0x00000000U);
-	psu_mask_write(0xFD080680, 0xFFFFFFFFU, 0x008AAA58U);
+	psu_mask_write(0xFD080680, 0xFFFFFFFFU, 0x0089EA58U);
 	psu_mask_write(0xFD080684, 0xFFFFFFFFU, 0x000079DDU);
 	psu_mask_write(0xFD080694, 0xFFFFFFFFU, 0x01E10210U);
 	psu_mask_write(0xFD080698, 0xFFFFFFFFU, 0x01E10000U);
@@ -340,27 +340,27 @@ static unsigned long psu_ddr_init_data(void)
 	psu_mask_write(0xFD080F14, 0xFFFFFFFFU, 0x09094F4FU);
 	psu_mask_write(0xFD080F18, 0xFFFFFFFFU, 0x09092B2BU);
 	psu_mask_write(0xFD081400, 0xFFFFFFFFU, 0x2A019FFEU);
-	psu_mask_write(0xFD081404, 0xFFFFFFFFU, 0x01100000U);
+	psu_mask_write(0xFD081404, 0xFFFFFFFFU, 0x02120000U);
 	psu_mask_write(0xFD08141C, 0xFFFFFFFFU, 0x01264300U);
 	psu_mask_write(0xFD08142C, 0xFFFFFFFFU, 0x00041800U);
 	psu_mask_write(0xFD081430, 0xFFFFFFFFU, 0x70800000U);
 	psu_mask_write(0xFD081440, 0xFFFFFFFFU, 0x2A019FFEU);
-	psu_mask_write(0xFD081444, 0xFFFFFFFFU, 0x01100000U);
+	psu_mask_write(0xFD081444, 0xFFFFFFFFU, 0x02120000U);
 	psu_mask_write(0xFD08145C, 0xFFFFFFFFU, 0x01264300U);
 	psu_mask_write(0xFD08146C, 0xFFFFFFFFU, 0x00041800U);
 	psu_mask_write(0xFD081470, 0xFFFFFFFFU, 0x70800000U);
 	psu_mask_write(0xFD081480, 0xFFFFFFFFU, 0x2A019FFEU);
-	psu_mask_write(0xFD081484, 0xFFFFFFFFU, 0x01100000U);
+	psu_mask_write(0xFD081484, 0xFFFFFFFFU, 0x02120000U);
 	psu_mask_write(0xFD08149C, 0xFFFFFFFFU, 0x01264300U);
 	psu_mask_write(0xFD0814AC, 0xFFFFFFFFU, 0x00041800U);
 	psu_mask_write(0xFD0814B0, 0xFFFFFFFFU, 0x70800000U);
 	psu_mask_write(0xFD0814C0, 0xFFFFFFFFU, 0x2A019FFEU);
-	psu_mask_write(0xFD0814C4, 0xFFFFFFFFU, 0x01100000U);
+	psu_mask_write(0xFD0814C4, 0xFFFFFFFFU, 0x02120000U);
 	psu_mask_write(0xFD0814DC, 0xFFFFFFFFU, 0x01264300U);
 	psu_mask_write(0xFD0814EC, 0xFFFFFFFFU, 0x00041800U);
 	psu_mask_write(0xFD0814F0, 0xFFFFFFFFU, 0x70800000U);
 	psu_mask_write(0xFD081500, 0xFFFFFFFFU, 0x15019FFEU);
-	psu_mask_write(0xFD081504, 0xFFFFFFFFU, 0x21100000U);
+	psu_mask_write(0xFD081504, 0xFFFFFFFFU, 0x22120000U);
 	psu_mask_write(0xFD08151C, 0xFFFFFFFFU, 0x01266300U);
 	psu_mask_write(0xFD08152C, 0xFFFFFFFFU, 0x00041800U);
 	psu_mask_write(0xFD081530, 0xFFFFFFFFU, 0x70400000U);
-- 
2.17.1

